test/services/katello/pulp/repository/yum_test.rb

```diff
@@ -132,6 +132,19 @@ def test_create_custom
         ensure
           delete_repo(@custom)
         end
+
+        def test_refresh
+          @repo = katello_repositories(:fedora_17_x86_64)
+          RepositorySupport.create_and_sync_repo(@repo)
+          @repo.root.product.stubs(:certificate).returns('repo_cert')
+          @repo.root.product.stubs(:key).returns('repo_key')
+          Katello::Repository.stubs(:feed_ca_cert).returns('ca_cert')
+
+          repo = Katello::Pulp::Repository::Yum.new(@repo, @master)
+          response = repo.refresh
+          assert_equal repo.backend_data["importers"].count, 1
+          assert_equal repo.backend_data["distributors"].count, response[1].count
```

Comment:
could you do something like delete a distributor manually, or modify some value that should result in an updated importer/distributor and then check that it is what is expected?